src/enhance.ts

```diff
@@ -16,6 +16,9 @@ code={\`
 $code\`}
 />`;
 
+const headerRegex = /(---)(.*---)/gms;
```

Comment:
Regex shouldn't be global (`g`) as only the first match should be replaced

Reply:
meanwhile, after I found some parsing issues, I changed it to `/---[^]*?(---)/`